### PR TITLE
bic: init at 1.0.0

### PR DESCRIPTION
**Pull request by hexagonal-sun**
###### Motivation for this change

Add a nix expression for the `bic` c interpreter and API explorer.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/bic/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  meta = {
+    description = "A C interpreter and API explorer";
+    longDescription = ''
+    bic This a project that allows developers to explore and test C-APIs using a
+    read eval print loop, also known as a REPL.
+    '';
+    license = with lib.licenses; [ gpl2Plus ];
+    homepage = "https://github.com/hexagonal-sun/bic";
+    platforms = with lib.platforms; unix;
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/bic/default.nix:22]:
```suggestion
  meta = with lib; {
    description = "A C interpreter and API explorer";
    longDescription = ''
      bic This a project that allows developers to explore and test C-APIs using a read eval print loop, also known as a REPL.
    '';
    license = with licenses; [ gpl2Plus ];
    homepage = "https://github.com/hexagonal-sun/bic";
    platforms = platforms.unix;
```

File: pkgs/development/interpreters/bic/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+stdenv.mkDerivation rec {
+  pname = "bic";
+  version = "1.0.0";
+
+  src = fetchurl {
+    url = "https://github.com/hexagonal-sun/${pname}/releases/download/v${version}/${pname}-v${version}.tar.gz";
+    sha256 = "553324e39d87df59930d093a264c14176d5e3aaa24cd8bff276531fb94775100";
+  };
+
+  buildInputs = [ readline gmp flex bison];
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/bic/default.nix:12]:
```suggestion
  nativeBuildInputs = [ flex bison ];

  buildInputs = [ readline gmp ];
```

File: pkgs/development/interpreters/bic/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{lib, stdenv, fetchurl, readline, gmp, flex, bison}:
+
+stdenv.mkDerivation rec {
+  pname = "bic";
+  version = "1.0.0";
+
+  src = fetchurl {
+    url = "https://github.com/hexagonal-sun/${pname}/releases/download/v${version}/${pname}-v${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/bic/default.nix:8]: Can we fetch the git repo instead?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129678` at 07a4f1dd run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>bic</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

flex is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/interpreters/bic/default.nix:12:3:
```
   |
12 |   buildInputs = [ readline gmp flex bison];
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/interpreters/bic/default.nix:14:3:
```
   |
14 |   meta = {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

bison is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/interpreters/bic/default.nix:12:3:
```
   |
12 |   buildInputs = [ readline gmp flex bison];
   |   ^
```
</li>
  </ul>
</details>

**Comment by hexagonal-sun** [comment]
@SuperSandro2000 Thanks for the review.  Comments have been addressed.

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -4207,6 +4207,12 @@
     githubId = 131599;
     name = "Martin Weinelt";
   };
+  hexagonal-sun = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:4210]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

**Comment by hexagonal-sun** [comment]
Bump.  If this could be merged, that would be great, Thanks!

File: pkgs/development/interpreters/bic/default.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    autoconf-archive
+    bison
+    flex
+  ];
+
+  meta = with lib; {
+    description = "A C interpreter and API explorer";
+    longDescription = ''
+    bic This a project that allows developers to explore and test C-APIs using a
+    read eval print loop, also known as a REPL.
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/bic/default.nix:35]:
```suggestion
    longDescription = ''
      bic This a project that allows developers to explore and test C-APIs using a
      read eval print loop, also known as a REPL.
```

Comment (hexagonal-sun) [reply pkgs/development/interpreters/bic/default.nix]: Thanks for the comment.  Fixed.
